### DISCUSSION TITLE
Is LibreChat considering launching a “lite” version?

### DISCUSSION THREAD

**Discussion by kuangxiaoye** [danny-avila/LibreChat#2198] [Feature Requests & Suggestions]
The current state of LibreChat has completely detached from its original simplicity and beauty, entering a phase of iteration for the sake of iteration.

Many features are unnecessary, and the emergence of services like Lobechat confirms this.

Does LibreChat really, truly need all these features?

Have we deviated from the essence of "conversation"?

**Comment by danny-avila** [comment]
>  entering a phase of iteration for the sake of iteration.

It's not for the sake of iteration. This app is built off community needs/requests

> and the emergence of services like Lobechat confirms this.

Lobechat offers more features though? And every active Web UI project continues to evolve as AI advances.

> Does LibreChat really, truly need all these features?

You should read the 2024 roadmap. I've gotten a lot of responses confirming the ask for the items:

https://librechat.ai/blog/2024-roadmap

Also if your concern is the added side panel on the right, you can easily hide it:

![image](https://github.com/danny-avila/LibreChat/assets/110412045/27e5c3cb-5c4f-480b-aa5d-812c36eb5d6f)

**Comment by danny-avila** [comment]
FWIW I think a "lite" version is not out of the question, but I can't imagine what this looks like exactly, aside from just being the base image, no RAG/meilisearch.

**Comment by danorlando** [comment]
@kuangxiaoye Your feedback is valued and appreciated and at the same time, I think it would be much more valuable if we were able to obtain some background information as to what led you to the conclusion that there needs to be a lite version. For us to have a truly meaningful conversation around this we would need to better understand your use case or what you were trying to accomplish with it that you could not because of the way in which the project is evolving.

**Comment by kuangxiaoye** [comment]
I am an independent developer and self-media practitioner. I am responsible for providing Chinese users with services about the world's top AI, and the birth of LibreChat is simply miraculous, as it can complete this task very effortlessly.I started following and using LibreChat when it was released last year.

But things started to get complicated. I was continuously following the main branch to iterate LibreChat, and more and more users began to complain：

“We just want to chat.”

This reminds me that looking back at the original LibreChat, it provided a very simple and efficient chat interface. But with the constant updates, I found that he was increasingly deviating from his original intention. (I have seen the efforts of developers, and the number and frequency of commit branches is very high, which is worthy of encouragement!)

However, I think LibreChat is somewhat overly comprehensive. It always wants to do all the work, adapting to any updates in language models. But, we should be able to notice that works like [one-api](https://github.com/songquanpeng/one-api), [chatnio](https://chatnio.net/) are gradually emerging, and they actually put more effort into the iteration of endpoints! They merge multiple models, and sometimes, just by adding a custom model name, a new model can be integrated. They are also able to calculate the number of tokens more accurately.In terms of user interaction, work like [lobechat](https://chat-preview.lobehub.com/) stands out even more. They eliminate the need for complex user and token calculation systems, placing the focus on a simple "chat" interface. For other tasks, developers can opt to integrate projects like one-api, which are more suited for these purposes.

Perhaps the positioning of LibreChat itself is different, aiming to do many things. However, I've received some genuine feedback from forums and group chats, where people are surprised by the "enormous" nature of LibreChat. Such as When people see the Docker container up and running, it's hard to believe that this is the kind of size a "web page" should have.

LibreChat is a great project anyway. I hope to let the excellent developers of LibreChat hear the voice of a user who has been using LibreChat since its birth. I hope the project gets better and better！

**Reply by danny-avila** [reply]
> However, I've received some genuine feedback from forums and group chats, where people are surprised by the "enormous" nature of LibreChat. Such as When people see the Docker container up and running, it's hard to believe that this is the kind of size a "web page" should have.

On this point, I think it's not fair to point to call it "enormous" when the docker setup is to provide ease of installation/deployment, and therefore includes all dev dependencies to build the data-provider package and the frontend code, along with node/npm fully installed, and not to mention mongo and meilisearch. It is not using the most optimized build steps for the storage-sensitive.

I did a bit of an experiment and here are the results.

- Bare node:18-alpine is 166 MB
- The compiled client-side code is 5.62 MB
- The compiled data-provider package is 896 KB (63.9 kB gzipped) but let's round up to 1 MB
- The backend server is 550 MB which I wager is average for a full-stack, multi-user app.

So we can optimize a docker image to be 672.62 MB in size with the bare minimum, no dev dependencies.

The non-optimized image size for LibreChat is 454.13 MB compressed, about 1.5 GB unpacked.

The more-but-not-fully optimized image size (the "api" image) is smaller at 360.25 MB compressed.

lobechat is 120.29 MB with no multi-user support or database dependencies.

So yes I think we can make a "lite" version that makes the base LibreChat image smaller (more than half the current size) but it will take trial and error along with docker expertise to fully make this possible. To truly make this "lighter" would be to use remote services for mongodb and/or opt out of meilisearch and RAG features.

Perhaps it's possible to split an imge by workspace into just the output code and none of the dependencies, I will investigate this. I'm attempting this already via the "api" image but I'm sure I could be doing something better.

---

### screenshots

![image](https://github.com/danny-avila/LibreChat/assets/110412045/0e89906d-fe48-4ca3-b465-ec39b721350e)
![image](https://github.com/danny-avila/LibreChat/assets/110412045/b7f1af81-caef-4d28-a56c-353b63a20212)
![image](https://github.com/danny-avila/LibreChat/assets/110412045/cd33097a-9129-4ff8-b0de-a1682e331964)
![image](https://github.com/danny-avila/LibreChat/assets/110412045/7b1ce37d-1e00-4369-8f85-ef48e0b8883e)

**Comment by danorlando** [comment]
Thanks for the response and feedback. This is certainly more along the lines of the type of thing that could lead to something actionable. I'm unsurprised by the sentiment tbh. I had some similar feelings initially when I came back to the project after being away for a few months. Scaling apps like this is truly an art because you have several camps of users: "power users" that want it to be feature rich with the latest and greatest tech, institutions that want to host it with their own tech stack and infrastructure, and individuals who value simplicity without all the bells and whistles to get in the way. While measures have been taken to help simplify the configuration process, there might be some merit to considering other opportunities like a CLI tool for example, which allows for a custom docker build based on a series of feature flags that are provided. In this way, a user could specify exactly what features they want and just leave out the rest. Obviously that is purely hypothetical and I'm just thinking out loud here, but the point is that there might be opportunities to support these different user groups by further obfuscating the complexities of the application. This to me seems like a more feasible solution than supporting a separate "lite" version.

**Reply by kuangxiaoye** [reply]
Your suggestion sounds better! This is what I want to say!

**Comment by danny-avila** [comment]
I think this update will appease the main concern, adding a lot of configuration without changing code, while only showing the user a simple interface:

https://github.com/danny-avila/LibreChat/issues/1617

**Comment by danny-avila** [comment]
> LibreChat is a great project anyway. I hope to let the excellent developers of LibreChat hear the voice of a user who has been using LibreChat since its birth. I hope the project gets better and better！

By the way I hope I don't come across as off-putting. I deeply appreciate your feedback and I think you single-handedly have influenced the development direction of LibreChat, both past and present. I only put as much attention and detail to this discussion as I deeply care to address any grievances or concerns users have with LibreChat. Thanks again for sparking this discussion.

**Reply by kuangxiaoye** [reply]
No worries, throughout this nearly one-year iteration, I have raised issues and communicated with you multiple times, and you have patiently provided answers. Your concerns are unnecessary. I am very grateful to you for initiating and maintaining the LibreChat project.This is just a feasibility discussion. I hope it will be of some help to the future development of the project.
